### PR TITLE
Upgrade to 20.2

### PR DESCRIPTION
**Pull request by sodre**
Mostly I am trying to figure out a way out of this 
  circle-ci bug/issue

### CODE REVIEW

File: ci_support/run_docker_build.sh [hunk 1]
```diff
@@ -41,10 +58,16 @@ conda clean --lock
...
+touch /feedstock_root/build_artefacts/conda-forge-build-done
 EOF
+
+echo "result from docker run: $?"
+
+# double-check that the build got to the end
+# see https://github.com/conda-forge/conda-smithy/pull/337
+# for a possible fix
+set -x
+test -f "$FEEDSTOCK_ROOT/build_artefacts/conda-forge-build-done" || exit 1
```
Comment (scopatz) [inline ci_support/run_docker_build.sh]: I don't think that you need this last line here

Comment (jakirkham) [reply ci_support/run_docker_build.sh]: That's autogenerated by `conda-smithy`. Please see PR ( https://github.com/conda-forge/conda-smithy/pull/337 ) for details.

Comment (scopatz) [reply ci_support/run_docker_build.sh]: Oh! Sorry you are right, I was looking at the wrong file

Comment (jakirkham) [reply ci_support/run_docker_build.sh]: No worries. It is a little strange, but there is a motivating use case for it. Maybe we can tweak the comment above to be more clear?

**Comment by sodre** [comment]
It does not. I was adding another command just to make sure it was not something else going on with the command before. 

I even tried to remove the tests altogether.

**Comment by scopatz** [comment]
Hmm weird. Any ideas @conda-forge/core?

**Comment by isuruf** [comment]
Can you try printing the error code after each of the commands in build.sh?

**Comment by sodre** [comment]
@isuruf 
Everything in build.sh returns 0. You can see that the TEST phase also passes and the package is "ready" to be uploaded.

```
TEST START: /feedstock_root/build_artefacts/linux-64/erlang-20.2-0.tar.bz2
Deleting work directory, /feedstock_root/build_artefacts/erlang_1518407316937/work/otp-OTP-20.2

The following NEW packages will be INSTALLED:

    ca-certificates: 2018.1.18-0 conda-forge
    erlang:          20.2-0      local      
    ncurses:         5.9-10      conda-forge
    openssl:         1.0.2n-0    conda-forge
    readline:        7.0-0       conda-forge
    zlib:            1.2.11-0    conda-forge


WARNING: conda-build appears to be out of date. You have version 2.1.18 but the
latest version is 3.4.1. Run

conda update -n root conda-build

to get the latest version.

+ source /opt/conda/bin/activate /feedstock_root/build_artefacts/erlang_1518407316937/_t_env
+ /bin/bash -x -e /feedstock_root/build_artefacts/erlang_1518407316937/test_tmp/run_test.sh
+ erl -version
Erlang (SMP,ASYNC_THREADS,HIPE) (BEAM) emulator version 9.2

+ erlc
TEST END: /feedstock_root/build_artefacts/linux-64/erlang-20.2-0.tar.bz2
INFO:conda_build.config:--dirty flag and --keep-old-work not specified.Removing build/test folder after successful build/test.

# Automatic uploading is disabled
# If you want to upload package(s) to anaconda.org later, type:

anaconda upload /feedstock_root/build_artefacts/linux-64/erlang-20.2-0.tar.bz2

# To have conda build upload to anaconda.org automatically, use
# $ conda config --set anaconda_upload yes

anaconda_upload is not set.  Not uploading wheels: []
result from docker run: 0
```

You can confirm this by building it by hand using the condaforge/linux-anvil container.

File: ci_support/run_docker_build.sh [hunk 2]
```diff
@@ -41,10 +58,19 @@ conda clean --lock
...
 
-# Embarking on 1 case(s).
-    set -x
-    export CONDA_PERL=5.20.3.1
-    set +x
-    conda build /recipe_root --quiet || exit 1
-    upload_or_check_non_existence /recipe_root conda-forge --channel=main || exit 1
+set +e
+echo "[REMOVE] calling conda build on recipe_root"
+conda build /recipe_root --debug --quiet || echo "[REMOVE] conda build rv: \$?"
```
Comment (sodre) [inline ci_support/run_docker_build.sh]: This last "[REMOVE] ..." message never shows up on the screen.

Running conda build on on --debug mode also does not show anything right away.

I am really out of ideas here, any help is appreciated.

cc: @isuruf, @jakirkham

File: ci_support/run_docker_build.sh [hunk 3]
```diff
@@ -58,7 +58,9 @@ conda clean --lock
 conda install --yes --quiet conda-forge-build-setup
 source run_conda_forge_build_setup
 
+conda install --yes conda-build=3
 conda build /recipe_root --quiet || exit 1
+conda install --yes conda-build=2
```
Comment (sodre) [inline ci_support/run_docker_build.sh]: update_or_check needs conda-build=2

File: ci_support/run_docker_build.sh [hunk 4]
```diff
@@ -58,7 +58,9 @@ conda clean --lock
 conda install --yes --quiet conda-forge-build-setup
 source run_conda_forge_build_setup
 
+conda install --yes conda-build=3
```
Comment (sodre) [inline ci_support/run_docker_build.sh]: @jakirkham @scopatz 
It looks like using conda-build=3 fixes the post build error we were having.

Is that an okay temporary solution to get the package updated?

Comment (scopatz) [reply ci_support/run_docker_build.sh]: Yes I think that is fine. We are moving to conda-build=3 anyway

**Pull request merged**